app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-link-click-tracker.js

```diff
@@ -0,0 +1,156 @@
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+  GOVUK.analyticsGA4 = GOVUK.analyticsGA4 || {}
+
+  GOVUK.analyticsGA4.linkClickTracker = {
+
+    trackLinkClicks: function () {
+      this.internalLinksDomain = 'www.gov.uk/'
+      document.querySelector('body').addEventListener('click', this.handleClick.bind(this))
+      document.querySelector('body').addEventListener('contextmenu', this.handleClick.bind(this))
+      document.querySelector('body').addEventListener('mousedown', this.handleMousedown.bind(this))
+    },
+
+    handleClick: function (event) {
+      var element = event.target
+
+      if (element.tagName !== 'A') {
+        element = element.closest('a')
+      }
+
+      if (!element) {
+        return
+      }
+
+      var clickData = {}
+      var href = element.getAttribute('href')
+
+      if (this.isMailToLink(href)) {
+        clickData.type = 'email'
+        clickData.external = 'true'
+      } else if (this.isDownloadLink(href)) {
+        clickData.type = 'download'
+        clickData.external = 'false'
+      } else if (this.isExternalLink(href)) {
+        clickData.type = 'generic link'
```

Comment:
@andysellick Yep this is from the implementation guide. I'm guessing this is because compared to a mailto and download links, these are "generic links" - but I agree that external link would make more sense.